mmdet3d/models/fusion_layers/vote_fusion.py

```diff
@@ -0,0 +1,243 @@
+        img_norm_cfg (dict): image normalization config.
+        max_imvote_per_pixel (int): max number of imvotes.
+    """
+
+    def __init__(self, num_classes=10, max_imvote_per_pixel=3):
+        super(VoteFusion, self).__init__()
+        self.num_classes = num_classes
+        self.max_imvote_per_pixel = max_imvote_per_pixel
+
+    def forward(self,
+                imgs,
+                bboxes_2d_rescaled,
+                seeds_3d_depth,
+                img_metas,
+                calibs,
+                eps=1e-6):
+        """Forward function.
+
+        Args:
+            imgs (list[torch.Tensor]): Image features.
+            bboxes_2d_rescaled (list[torch.Tensor]): 2D bboxes.
+            seeds_3d_depth (torch.Tensor): 3D seeds.
+            img_metas (list[dict]): Meta information of images.
+            calibs: Camera calibration information of the images.
+
+        Returns:
+            torch.Tensor: Concatenated cues of each point.
+            torch.Tensor: Validity mask of each feature.
+        """
+        img_features = []
+        masks = []
+        for i, data in enumerate(
+                zip(imgs, bboxes_2d_rescaled, seeds_3d_depth, img_metas)):
+            img, bbox_2d_rescaled, seed_3d_depth, img_meta = data
+            bbox_num = bbox_2d_rescaled.shape[0]
+            seed_num = seed_3d_depth.shape[0]
+
+            bbox_2d_origin = bbox_2d_rescaled.clone().float()
+            img_shape = img_meta['img_shape']
+            ori_shape = img_meta['ori_shape']
+            # origin bboxes
+            img_h, img_w, _ = img_shape
+            ori_h, ori_w, _ = ori_shape
+
+            img_scale_factor = (
+                seed_3d_depth.new_tensor(img_meta['scale_factor'][:2])
+                if 'scale_factor' in img_meta else [1.0, 1.0])
+            img_flip = img_meta['flip'] if 'flip' in img_meta else False
+            img_crop_offset = (
+                seed_3d_depth.new_tensor(img_meta['img_crop_offset'])
+                if 'img_crop_offset' in img_meta else 0)
+
+            # first reverse the data transformations
+            xyz_depth = apply_3d_transformation(
+                seed_3d_depth, 'DEPTH', img_meta, reverse=True)
+
+            # then convert from depth coords to camera coords
+            xyz_cam = Coord3DMode.convert_point(
+                xyz_depth,
+                Coord3DMode.DEPTH,
+                Coord3DMode.CAM,
+                rt_mat=calibs['Rt'][i])
+
+            # project to 2d to get image coords (uv)
+            uv_origin = points_cam2img(xyz_cam, calibs['K'][i])
+            uv_origin = (uv_origin - 1).round()
+
+            # rescale uv coordinates
+            uv_rescaled = uv_origin.clone().float()
+            uv_rescaled[:, 0] = uv_rescaled[:, 0] * img_scale_factor[0]
+            uv_rescaled[:, 1] = uv_rescaled[:, 1] * img_scale_factor[1]
+            uv_rescaled += img_crop_offset
+
+            # flip uv coordinates and bbox
+            if img_flip:
```

Comment:
Can we also implement small utils functions to flip/scale/ image coordinates? Thus they can have more general usages.

Reply:
Updated a simple implementation. Added TODO